### ISSUE TITLE
Build Android project but opencv links wrongly

### ISSUE THREAD

**Issue by xv994** [google/mediapipe#4133]
### OS Platform and Distribution

Ubuntu 22.04

### Programming Language and version

java 17

### Installed using virtualenv? pip? Conda?(if python)

conda

### Bazel version

5.2.0

### Android SDK and NDK versions (if android)

SDK is 31 and NDK is 21

### Android AAR (if android)

None

### OpenCV version (if running on desktop)

4

### Describe the problem

```shell
when I run the sh "bazel build -c opt --config=android_arm64 $APPLICATION_PATH:helloworld" to build my android basic project. There was something wrong, like this: "ERROR: /home/xvzh/mediapipe/third_party/BUILD:186:6: no such target '@android_opencv//:libopencv_arm64-v8a': target 'libopencv_arm64-v8a' not declared in package '' defined by /home/xvzh/.cache/bazel/_bazel_xvzh/2826d3a0ece8a58c8bc8a80083946c30/external/android_opencv/BUILD.bazel and referenced by '//third_party:opencv_binary' ERROR: Analysis of target '//mediapipe/examples/android/src/java/com/google/mediapipe/apps/basic:helloworld' failed; build aborted: ".
I think I wrongly built my BUILD or WORKSPACE, but I cannot find the tutor about opencv with android. So, I am here to find you guys' help. Thank you for your time.
```

### Complete Logs

```shell
xvzh@xvzh:~/mediapipe$ bazel build -c opt --config=android_arm64 mediapipe/examples/android/src/java/com/google/mediapipe/apps/basic:helloworld
DEBUG: /home/xvzh/.cache/bazel/_bazel_xvzh/2826d3a0ece8a58c8bc8a80083946c30/external/org_tensorflow/third_party/repo.bzl:132:14: 
Warning: skipping import of repository 'com_google_absl' because it already exists.
DEBUG: /home/xvzh/.cache/bazel/_bazel_xvzh/2826d3a0ece8a58c8bc8a80083946c30/external/org_tensorflow/third_party/repo.bzl:132:14: 
Warning: skipping import of repository 'com_google_benchmark' because it already exists.
DEBUG: /home/xvzh/.cache/bazel/_bazel_xvzh/2826d3a0ece8a58c8bc8a80083946c30/external/org_tensorflow/third_party/repo.bzl:132:14: 
Warning: skipping import of repository 'flatbuffers' because it already exists.
DEBUG: /home/xvzh/.cache/bazel/_bazel_xvzh/2826d3a0ece8a58c8bc8a80083946c30/external/org_tensorflow/third_party/repo.bzl:132:14: 
Warning: skipping import of repository 'pybind11_bazel' because it already exists.
DEBUG: /home/xvzh/.cache/bazel/_bazel_xvzh/2826d3a0ece8a58c8bc8a80083946c30/external/org_tensorflow/third_party/repo.bzl:132:14: 
Warning: skipping import of repository 'com_googlesource_code_re2' because it already exists.
DEBUG: /home/xvzh/.cache/bazel/_bazel_xvzh/2826d3a0ece8a58c8bc8a80083946c30/external/org_tensorflow/third_party/repo.bzl:132:14: 
Warning: skipping import of repository 'com_google_protobuf' because it already exists.
DEBUG: /home/xvzh/.cache/bazel/_bazel_xvzh/2826d3a0ece8a58c8bc8a80083946c30/external/org_tensorflow/third_party/repo.bzl:132:14: 
Warning: skipping import of repository 'com_google_googletest' because it already exists.
DEBUG: /home/xvzh/.cache/bazel/_bazel_xvzh/2826d3a0ece8a58c8bc8a80083946c30/external/org_tensorflow/third_party/repo.bzl:132:14: 
Warning: skipping import of repository 'com_github_gflags_gflags' because it already exists.
DEBUG: /home/xvzh/.cache/bazel/_bazel_xvzh/2826d3a0ece8a58c8bc8a80083946c30/external/org_tensorflow/third_party/repo.bzl:132:14: 
Warning: skipping import of repository 'zlib' because it already exists.
DEBUG: /home/xvzh/.cache/bazel/_bazel_xvzh/2826d3a0ece8a58c8bc8a80083946c30/external/org_tensorflow/third_party/repo.bzl:132:14: 
Warning: skipping import of repository 'rules_python' because it already exists.
DEBUG: /home/xvzh/.cache/bazel/_bazel_xvzh/2826d3a0ece8a58c8bc8a80083946c30/external/org_tensorflow/third_party/repo.bzl:132:14: 
Warning: skipping import of repository 'build_bazel_rules_apple' because it already exists.
DEBUG: /home/xvzh/.cache/bazel/_bazel_xvzh/2826d3a0ece8a58c8bc8a80083946c30/external/org_tensorflow/third_party/repo.bzl:132:14: 
Warning: skipping import of repository 'build_bazel_rules_swift' because it already exists.
DEBUG: /home/xvzh/.cache/bazel/_bazel_xvzh/2826d3a0ece8a58c8bc8a80083946c30/external/org_tensorflow/third_party/repo.bzl:132:14: 
Warning: skipping import of repository 'build_bazel_apple_support' because it already exists.
DEBUG: /home/xvzh/.cache/bazel/_bazel_xvzh/2826d3a0ece8a58c8bc8a80083946c30/external/org_tensorflow/third_party/repo.bzl:132:14: 
Warning: skipping import of repository 'xctestrunner' because it already exists.
DEBUG: /home/xvzh/.cache/bazel/_bazel_xvzh/2826d3a0ece8a58c8bc8a80083946c30/external/org_tensorflow/third_party/repo.bzl:132:14: 
Warning: skipping import of repository 'pybind11' because it already exists.
DEBUG: /home/xvzh/.cache/bazel/_bazel_xvzh/2826d3a0ece8a58c8bc8a80083946c30/external/org_tensorflow/third_party/repo.bzl:132:14: 
Warning: skipping import of repository 'com_github_glog_glog' because it already exists.
WARNING: /home/xvzh/mediapipe/mediapipe/framework/BUILD:54:24: in cc_library rule //mediapipe/framework:calculator_cc_proto: target '//mediapipe/framework:calculator_cc_proto' depends on deprecated target '@com_google_protobuf//:cc_wkt_protos': Only for backward compatibility. Do not use.
WARNING: /home/xvzh/mediapipe/mediapipe/framework/tool/BUILD:184:24: in cc_library rule //mediapipe/framework/tool:field_data_cc_proto: target '//mediapipe/framework/tool:field_data_cc_proto' depends on deprecated target '@com_google_protobuf//:cc_wkt_protos': Only for backward compatibility. Do not use.
ERROR: /home/xvzh/mediapipe/third_party/BUILD:186:6: no such target '@android_opencv//:libopencv_arm64-v8a': target 'libopencv_arm64-v8a' not declared in package '' defined by /home/xvzh/.cache/bazel/_bazel_xvzh/2826d3a0ece8a58c8bc8a80083946c30/external/android_opencv/BUILD.bazel and referenced by '//third_party:opencv_binary'
ERROR: Analysis of target '//mediapipe/examples/android/src/java/com/google/mediapipe/apps/basic:helloworld' failed; build aborted: 
INFO: Elapsed time: 51.518s
INFO: 0 processes.
FAILED: Build did NOT complete successfully (4 packages loaded, 858 targets co\
nfigured)
```

**Comment by ayushgdev** [comment]
Hello @xv994 Yes this is a setup issue. Few suggestions for changes:
- Use Android SDK 30.0.3 and Android NDK 21.1.6352462 (install via Android Studio)
- Make [this change](https://github.com/google/mediapipe/issues/3719#issuecomment-1345868251) in the WORKSPACE file 
- Make sure you followed each step of [android setup guide ](https://google.github.io/mediapipe/getting_started/android.html)

**Comment by xv994** [comment]
Hey guys, thank you for your suggestions. I recloned the repository and edited the WORKSPACE as your forth suggestion, then I built the basic project successfully. In addition, when the build_tools_version is 34.0.0-rc1, it reported that it miss dx.jar, error information just like this " missing input file 'external/androidsdk/build-tools/34.0.0-rc1/lib/dx.jar', owner: '@androidsdk//:build-tools/34.0.0-rc1/lib/dx.jar'". When I changed it to 30.0.3, error disappeared. Thank you for your help again!
